core/src/mill/eval/Evaluator.scala

```diff
@@ -191,6 +191,27 @@ case class Evaluator[T](home: Path,
         }
     }
   }
+
+  def destSegments(labelledTask : Labelled[_]) : Segments = {
```

Comment:
This is the core of the PR I guess. This makes sure that "foreign" modules do not conflict with local modules